### ISSUE TITLE
False positive in UselessDotAllModifierCheckStrategy

### ISSUE THREAD

**Issue by dryabov** [kalessil/phpinspectionsea#1440]
Current behaviour
--
The inspection is triggered on
`preg_replace('#\[t.e.s.t\]#s', '', $str);`

Expected behaviour
--
Escaped characters shouldn't be taken into account during "normalization".

Solution
--
Solution is to modify two following lines in `UselessDotAllModifierCheckStrategy.java`:
https://github.com/kalessil/phpinspectionsea/blob/f646279f56770f754343470d97dfa4098eaa17ce/src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/regularExpressions/modifiersStrategy/UselessDotAllModifierCheckStrategy.java#L30-L31

Either exclude escaped characters from normalized string:
```java
final String normalized = pattern.replaceAll("\\\\.", "").replaceAll("\\[[^\\]]+\\]", "");
if (normalized.indexOf('.') >= 0) {
```
or do check using single regex
```java
if (!pattern.matches("(\\\\.|\\[[^]]*\\]|[^\\[.])*")) {
```

**Comment by dryabov** [comment]
PS. Just updated the example.

**Comment by kalessil** [comment]
Confirmed, thank you for reporting Denis. I'll take care of fixing this.

**Comment by kalessil** [comment]
Fixed!
